core/store/src/trie/shard_tries.rs

```diff
@@ -91,18 +90,19 @@ impl ShardTries {
         Arc::ptr_eq(&self.0, &other.0)
     }
 
-    pub fn new_trie_update(&self, shard_uid: ShardUId, state_root: CryptoHash) -> TrieUpdate {
-        TrieUpdate::new(Rc::new(self.get_trie_for_shard(shard_uid)), state_root)
+    pub fn new_trie_update(&self, shard_uid: ShardUId, state_root: StateRoot) -> TrieUpdate {
+        TrieUpdate::new(std::rc::Rc::new(self.get_trie_for_shard(shard_uid, state_root)))
```

Comment:
I'd keep `Rc`: `Rc`, `Arc`, `Cell`, `RefCell`, `Cow` are vocabulary types which are usually used imported. And rust is generally much friendlier to using unqualified names than C++, as there isn't a problem with polluting the global namespace.